models/iaf_cond_alpha_mc.cpp

```diff
@@ -117,13 +117,39 @@ nest::iaf_cond_alpha_mc_dynamics( double, const double y[], double f[], void* pn
   assert( pnode );
   const nest::iaf_cond_alpha_mc& node = *( reinterpret_cast< nest::iaf_cond_alpha_mc* >( pnode ) );
 
+  bool is_refractory = false;
+
   // compute dynamics for each compartment
   // computations written quite explicitly for clarity, assume compile
   // will optimized most stuff away ...
   for ( size_t n = 0; n < N::NCOMP; ++n )
   {
-    // membrane potential for current compartment
-    const double V = y[ S::idx( n, S::V_M ) ];
+    // membrane potential for current compartment and coupling currents
+    double V;
+    double I_conn;
+    switch ( n )
+    {
+    case N::SOMA:
+    {
+      is_refractory = node.S_.r_ > 0;
+      V = is_refractory ? node.P_.V_reset : std::min( y[ S::idx( n, S::V_M ) ], node.P_.V_th );
+      I_conn = node.P_.g_conn[ n ] * ( V - y[ S::idx( n + 1, S::V_M ) ] );
+      break;
+    }
+    case N::PROX:
+    {
+      V = y[ S::idx( n, S::V_M ) ];
+      I_conn = node.P_.g_conn[ n - 1 ] * ( V - y[ S::idx( n - 1, S::V_M ) ] )
+        + node.P_.g_conn[ n ] * ( V - y[ S::idx( n + 1, S::V_M ) ] );
+      break;
+    }
+    case N::DIST:
+    {
+      V = y[ S::idx( n, S::V_M ) ];
+      I_conn = node.P_.g_conn[ n - 1 ] * ( V - y[ S::idx( n - 1, S::V_M ) ] );
+      break;
+    }
+    }
```

Comment:
I think the code will become clearer by using explicit indices (`n` has fixed values in the different cases anyways).
```suggestion
    double V;
    double I_conn;
    switch ( n )
    {
    case N::SOMA:
    {
      is_refractory = node.S_.r_ > 0;
      V = is_refractory ? node.P_.V_reset : std::min( y[ S::idx( N::SOMA, S::V_M ) ], node.P_.V_th );
      I_conn = node.P_.g_conn[ N::SOMA ] * ( V - y[ S::idx( N::PROX, S::V_M ) ] );
      break;
    }
    case N::PROX:
    {
      V = y[ S::idx( N::PROX, S::V_M ) ];
      I_conn = node.P_.g_conn[ N::SOMA ] * ( V - y[ S::idx( N::SOMA, S::V_M ) ] )
        + node.P_.g_conn[ N::PROX ] * ( V - y[ S::idx( N::DIST, S::V_M ) ] );
      break;
    }
    case N::DIST:
    {
      V = y[ S::idx( N::DIST, S::V_M ) ];
      I_conn = node.P_.g_conn[ N::PROX ] * ( V - y[ S::idx( N::PROX - 1, S::V_M ) ] );
      break;
    }
    }
```

Reply:
I'm agreed with the suggestion, but at this point, we've pretty much already unrolled the loop by hand, so we may as well remove the ``for`` and the ``switch`` entirely.

Reply:
Not quite, I'd say. We have the lines computing `I_syn_ex`, `I_syn_in`, `I_L` and the lines computing the five elements of `f` for each compartment which we would need to replicate for each compartment if we unrolled. Therefore, I think it better to keep the loop.